control/veda_deliver.py

```diff
@@ -127,17 +133,34 @@ def run(self):
         self._UPDATE_DATA()
         self._CLEANUP()
 
-        # Transcription Process
-        # We only want to generate transcripts for `desktop_mp4` profile.
-        if self.encode_profile == 'desktop_mp4' and self.video_query.process_transcription:
+        # We only want to generate transcripts when all the encodings(except for YT and Review) are done.
+        if utils.is_video_ready(self.video_query.edx_id, ignore_encodes=['review', 'youtube']):
```

Comment:
I think we should have this check inside `start_transcription` method.

Reply:
I was doing that before, it kinda feel dirty.